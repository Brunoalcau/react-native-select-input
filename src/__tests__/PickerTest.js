import React from 'react';
import { TouchableWithoutFeedback } from 'react-native';
import { shallow } from 'enzyme';
import colors from '../../fixtures/colors';
import Picker from '../Picker';

describe('Picker', () => {
  beforeEach(() => {
    const options = [{
      value: 'first-value',
      label: 'First value',
    }, {
      value: 'second-value',
      label: 'Second value',
    }];

    this.props = {
      label: 'label',
      options,
      value: options[0].value,
      colors,
      done: 'Done',
      onChange: jest.fn(),
      toggleShowOptions: jest.fn(),
    };
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Picker {...this.props} />);

    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders correctly with a testProperty set', () => {
    const wrapper = shallow(
      <Picker
        {...this.props}
        testProperty="foo"
      />,
    );

    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should show picker if modal is visible', () => {
    const wrapper = shallow(<Picker {...this.props} />, {
      lifecycleExperimental: true,
    });

    jest.useFakeTimers();

    expect(wrapper.state().showPicker).toBe(false);

    wrapper.setProps({
      visible: true,
    });

    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 150);
    setTimeout(() => expect(wrapper.state().showPicker).toBe(true), 0);
  });

  it('should close options if pressed', () => {
    const wrapper = shallow(
      <Picker
        {...this.props}
        visible
      />,
    );

    wrapper.find(TouchableWithoutFeedback).props().onPress();

    expect(this.props.toggleShowOptions).toHaveBeenCalled();
  });
});
