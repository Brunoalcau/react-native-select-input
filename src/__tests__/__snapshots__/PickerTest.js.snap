// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <TouchableWithoutFeedback
    accessible={false}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.1)",
          "borderColor": "red",
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <SlideUpAnimation
        onAnimationEnd={[Function]}
        style={null}
        visible={false}
      >
        <PickerHeader
          done="Done"
          doneStyle={null}
          testProperty={null}
          toggleShowOptions={[Function]}
        />
        <Picker
          mode="dialog"
          onValueChange={[Function]}
          selectedValue="first-value"
          style={
            Object {
              "backgroundColor": "#D9D9D9",
              "width": "100%",
            }
          }
        >
          <PickerItem
            key="first-value"
            label="First value"
            value="first-value"
          />
          <PickerItem
            key="second-value"
            label="Second value"
            value="second-value"
          />
        </Picker>
      </SlideUpAnimation>
    </View>
  </TouchableWithoutFeedback>
</Modal>
`;

exports[`Picker renders correctly with a testProperty set 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <TouchableWithoutFeedback
    accessible={false}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.1)",
          "borderColor": "red",
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <SlideUpAnimation
        onAnimationEnd={[Function]}
        style={null}
        visible={false}
      >
        <PickerHeader
          done="Done"
          doneStyle={null}
          testProperty="foo"
          toggleShowOptions={[Function]}
        />
        <Picker
          mode="dialog"
          onValueChange={[Function]}
          selectedValue="first-value"
          style={
            Object {
              "backgroundColor": "#D9D9D9",
              "width": "100%",
            }
          }
        >
          <PickerItem
            key="first-value"
            label="First value"
            value="first-value"
          />
          <PickerItem
            key="second-value"
            label="Second value"
            value="second-value"
          />
        </Picker>
      </SlideUpAnimation>
    </View>
  </TouchableWithoutFeedback>
</Modal>
`;
